Added queries and mutations for action items

NishantSinghhhhh opened the pull request:
What kind of change does this PR introduce?

Feature: Migrates ActionItem Category handling from Express endpoints to GraphQL schema and resolvers.

Issue Number: #3320

Snapshots/Videos:

N/A (No UI changes; these changes affect backend GraphQL schemas and resolvers.)
If relevant, did you update the documentation?

I have added/updated relevant documentation within the code and README where needed.
Summary

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for creating and updating action item categories via new GraphQL mutations.
  - Introduced paginated queries for fetching action items and action item categories by organization.
  - Action items now include an "allotted hours" field to represent estimated completion time.
  - Action item and category types now expose related entities (e.g., organization, creator, updater) as nested objects instead of flat ID fields.
  - Added detailed error handling and authorization checks to resolvers for action items and categories.
  - Added a new resolver to fetch action item categories associated with action items.

- **Bug Fixes**
  - Corrected and unified references to action item and category tables throughout the application.

- **Refactor**
  - Replaced scalar ID fields in GraphQL types with object references for improved clarity and data access.
  - Updated input types and schemas for mutations and queries to support new features and relationships.
  - Extracted resolver logic into standalone functions with explicit typing and improved error handling.
  - Simplified error handling in assignee resolver to return null instead of throwing.
  - Reorganized imports and exports to remove redundant re-exports and improve clarity.

- **Chores**
  - Updated environment configuration for Redis connection handling.
  - Migrated and enhanced sample data and test data to align with new schema.
  - Updated GraphQL document nodes and type declarations to reflect schema changes.

- **Tests**
  - Added comprehensive unit and integration tests for new resolvers, mutations, and pagination logic.
  - Removed obsolete tests related to deprecated queries and input types.
  - Added tests covering authentication, authorization, error handling, and data retrieval for action items and categories.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mpdesai90 commented:
Please fix the failing test.

NishantSinghhhhh commented:
@meetulr sir, @palisadoes sir should I add any more changes in here ??

meetulr commented on src/graphql/types/ActionItem/categoryId.ts.ts:
```diff
@@ -0,0 +1,54 @@
+import type { GraphQLContext } from "~/src/graphql/context";
```
`actionItemCategory.ts`

NishantSinghhhhh commented:
I am looking at the whole PR again and will write all changes I make in here

NishantSinghhhhh commented:
changing ActionItemCategorybyOrganization.ts and 'ActionItembyOrganization.ts` into paginated queries

NishantSinghhhhh commented:
changed file names 

`ActionItemCategory` --->  `ActionItemCategorybyOrganization.ts`  
`ActionItem.ts` into `ActionItembyOrganization.ts` 

also converted these queries into paginated queries

NishantSinghhhhh commented:
Changed file names 

`ActionItemCategory` ---->  `ActionItemCategorybyOrganization.ts`  
`ActionItem.ts`  ----> `ActionItembyOrganization.ts` 

also converted these queries into paginated queries

Added comments in all files I have added or edited

mpdesai90 commented:
Please fix the failing tests.

NishantSinghhhhh commented:
@meetulr sir 

Changed file names 

`ActionItemCategory` ---->  `ActionItemCategorybyOrganization.ts`  
`ActionItem.ts`  ----> `ActionItembyOrganization.ts` 

also converted these queries into paginated queries

Added comments in all files I have added or edited

I have made sure that this PR follow the way we write code in other queries and mutation

NishantSinghhhhh commented:
@meetulr sir

Changed file names

ActionItemCategory ----> ActionItemCategorybyOrganization.ts
ActionItem.ts ----> ActionItembyOrganization.ts

also converted these queries into paginated queries

Added comments in all files I have added or edited

I have made sure that this PR follow the way we write code in other queries and mutation

NishantSinghhhhh commented:
@meetulr sir

Changed file names

ActionItemCategory ----> ActionItemCategorybyOrganization.ts
ActionItem.ts ----> ActionItembyOrganization.ts

also converted these queries into paginated queries

Added comments in all files I have added or edited

I have made sure that this PR follow the way we write code in other queries and mutation

NishantSinghhhhh commented:
If this PR is good should I go with PR present in talawa-admin?

meetulr commented:
@NishantSinghhhhh is the code coverage of this patch 100%?

NishantSinghhhhh commented:
Sir I was just waiting for your call that this PR is good, 

I will begin adding tests by today and make the patch coverage as close to 100%

palisadoes commented:
Closing as abandoned
